packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -130,7 +140,12 @@ const { vScrollEnd, isScrollEnd } = useScrollEnd({
   offset: computed(() => props.lazyLoading?.scrollOffset),
 });
 
-const isEmpty = computed(() => props.options.length === 0);
+const isEmptyMessage = computed<FlattenedKeysOf<OnyxTranslations> | undefined>(
+  () =>
+    (props.options.length === 0 && "listbox.empty") ||
+    (props.withSearch && props.searchTerm && props.options.length === 0 && "listbox.noMatch") ||
```

Comment:
listbox.noMatch can never be reached because it checks for `props.options.length === 0` which is always returned earlier with "listbox.empty". Also imho I think the `&&` syntax is a bit hard to understand. What do you think about this?

```ts
const isEmptyMessage = computed(() => {
  if (props.options.length === 0) return t.value("listbox.empty");
  if (props.withSearch && props.searchTerm && props.options.length === 0) {
    return t.value("listbox.noMatch");
  }
  return undefined;
});
```

Reply:
Also I'd ususally prefer to directly return the translated message instead of the key

Reply:
fixed